metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1175,104 +1175,43 @@ def get_tables_for_dataset(
-                    table=table.table_id,
-                )
+            max_batch_size: int = (
+                self.config.number_of_datasets_process_in_batch
+                if not self.config.profiling.enabled
+                else self.config.number_of_datasets_process_in_batch_if_profiling_enabled
+            )
 
-                _, shard = BigqueryTableIdentifier.get_table_and_shard(
-                    table_identifier.raw_table_name()
-                )
-                table_name = table_identifier.get_table_name().split(".")[-1]
-
-                # Sharded tables look like: table_20220120
-                # For sharded tables we only process the latest shard and ignore the rest
-                # to find the latest shard we iterate over the list of tables and store the maximum shard id
-                # We only has one special case where the table name is a date `20220110`
-                # in this case we merge all these tables under dataset name as table name.
-                # For example some_dataset.20220110 will be turned to some_dataset.some_dataset
-                # It seems like there are some bigquery user who uses this non-standard way of sharding the tables.
-                if shard:
-                    if not sharded_tables.get(table_name):
-                        # When table is only a shard we use dataset_name as table_name
-                        sharded_tables[table_name] = table
-                        continue
-
-                    stored_table_identifier = BigqueryTableIdentifier(
-                        project_id=project_id,
-                        dataset=dataset_name,
-                        table=sharded_tables[table_name].table_id,
-                    )
-                    _, stored_shard = BigqueryTableIdentifier.get_table_and_shard(
-                        stored_table_identifier.raw_table_name()
-                    )
-                    # When table is none, we use dataset_name as table_name
-                    assert stored_shard
-                    if stored_shard < shard:
-                        sharded_tables[table_name] = table
-                    continue
-                elif str(table_identifier).startswith(
-                    self.config.temp_table_dataset_prefix
-                ):
-                    logger.debug(f"Dropping temporary table {table_identifier.table}")
-                    self.report.report_dropped(table_identifier.raw_table_name())
-                    continue
-                else:
-                    if (
-                        table.time_partitioning
-                        or "range_partitioning" in table._properties
-                    ):
-                        partitioned_table_count_in_this_batch += 1
-
-                    table_items[table.table_id] = table
-
-                if (
-                    len(table_items) % self.config.number_of_datasets_process_in_batch
-                    == 0
```

Comment:
We were already batching this even if there was no partitioning right? What is this PR changing that will fix the issue?

Reply:
Earlier, we added the list of sharded tables to the last batch and because of this the last batch could be way more than the max batch size. This is now fixed